tests/unit/change/models/changeModel.spec.ts

```diff
@@ -0,0 +1,131 @@
+import { OsmElementType } from '@map-colonies/node-osm-elements';
+
+import * as functions from '../../../../src/change/models/changeManager';
+import { ChangeManager } from '../../../../src/change/models/changeManager';
+import { TestDataBuilder } from '../../../common/testDataBuilder';
+import { FeatureType } from '../../../../src/change/models/geojsonTypes';
+import { allFeatureTypes } from '../../../common/constants';
+import { ParseOsmElementsError } from '../../../../src/change/models/errors';
+import { osmElementsMap, OsmApiToElement } from '../../../common/samples/sampleData';
+
+interface ICustomSpies {
+  getChange: jest.SpyInstance;
+  getElement: jest.SpyInstance;
+}
+
+const getAllFeatureCasesByAction = (action: Actions): [Actions, FeatureType][] => allFeatureTypes.map((featureType) => [action, featureType]);
+
+const caseTable: [Actions, FeatureType][] = [...getAllFeatureCasesByAction(Actions.MODIFY), ...getAllFeatureCasesByAction(Actions.DELETE)];
+
+let changeManager: ChangeManager;
+let testDataBuilder: TestDataBuilder;
+let featureToSpiesMap: Map<FeatureType, ICustomSpies>;
+
+describe('ChangeManager', () => {
+  beforeAll(function () {
+    testDataBuilder = new TestDataBuilder();
+
+    const getChangeFromPointSpy = jest.spyOn(osmChangeGenerator, 'getChangeFromPoint');
+    const getChangeFromLineSpy = jest.spyOn(osmChangeGenerator, 'getChangeFromLine');
+    const getChangeFromPolygonSpy = jest.spyOn(osmChangeGenerator, 'getChangeFromPolygon');
+    const getNodeSpy = jest.spyOn(functions, 'getNodeFromElements');
+    const getWaySpy = jest.spyOn(functions, 'getOsmWayFromElements');
+
+    featureToSpiesMap = new Map<FeatureType, ICustomSpies>([
+      ['Point', { getChange: getChangeFromPointSpy, getElement: getNodeSpy }],
+      ['LineString', { getChange: getChangeFromLineSpy, getElement: getWaySpy }],
+      ['Polygon', { getChange: getChangeFromPolygonSpy, getElement: getWaySpy }],
+    ]);
+  });
+  beforeEach(function () {
+    changeManager = new ChangeManager({ log: jest.fn() });
+  });
+  afterEach(function () {
+    jest.clearAllMocks();
+  });
+  describe('#generateChange', () => {
+    test.each(getAllFeatureCasesByAction(Actions.CREATE))(
+      'should return a create changeModel with tempOsmId',
+      (action: Actions, type: FeatureType) => {
+        const tempOsmIdSpy = jest.spyOn(functions, 'getTempOsmId');
+        const { getChange, getElement } = featureToSpiesMap.get(type) as ICustomSpies;
+        const { request, expectedResult: expectedChangeResult } = testDataBuilder.setAction(action).setGeojson(type).getTestData();
+        const { geojson, osmElements, externalId } = request;
+
+        const result = changeManager.generateChange(request.action, geojson, osmElements, externalId);
+
+        // spies
+        expect(getElement).not.toHaveBeenCalled();
+        expect(getChange).toHaveBeenCalledWith({ action: action, feature: geojson });
+        expect(tempOsmIdSpy).toHaveBeenCalled();
```

Comment:
This is white box, you should not test the function body.the output of your function indicates  if the functions did well. so you need to check only the result of the function.
input --> good or bad,
output --> works or not

Reply:
done.